six/modules/c++/six/source/Utilities.cpp

```diff
@@ -1198,3 +1199,20 @@ void six::getErrors(const ErrorStatistics* errorStats,
         }
     }
 }
+
+std::auto_ptr<Data> six::readXML(const std::string& xmlPath, DataType dataType)
+{
+    io::FileInputStream xmlFileStream(xmlPath);
+    xml::lite::MinidomParser treeBuilder;
+    treeBuilder.parse(xmlFileStream);
+    xmlFileStream.close();
+
+    std::auto_ptr<logging::Logger> log(new logging::NullLogger());
```

Comment:
Don't need to bother with a pointer here.  Just make
`logging::NullLogger logger`
and then send in `&logger`